stdlib/public/core/StringStorage.swift

```diff
@@ -97,11 +100,33 @@ private typealias CountAndFlags = _StringObject.CountAndFlags
 @_fixed_layout
 @usableFromInline
 final internal class _StringStorage: _AbstractStringStorage {
+#if arch(i386) || arch(arm)
+  // The total allocated storage capacity. Note that this includes the required
+  // nul-terminator
+  @nonobjc
+  @usableFromInline
+  internal var _realCapacity: Int
+
+  @nonobjc
+  @usableFromInline
+  internal var _count: Int
+
+  @nonobjc
+  @usableFromInline
+  internal var _flags: _StringObject.Flags
```

Comment:
We also want capacity flags, but maybe 32-bit just won't get those optimizations. If there's any padding between this and the trailing allocation, go ahead and add a _padding field.

Reply:
There is no padding, but we can easily bump the bias to 20. Do you expect to use >16 bits? If so, I'll just add another UInt16.

Reply:
I will, probably, but it's not worth putting that on 32-bit. 32-bit just won't get those optimizations sometime in the future

Reply:
The old 18-byte offset ruled out 16-byte alignment completely, so bumping to 20 may actually be helpful. Neither is good if malloc alignment is 8 or greater, though. :(

Reply:
We also have a pointer at the end as a second trailing allocation. For malloc estimates, check out (and perhaps update with 32-bit findings) https://gist.github.com/milseman/ec0096c8763dd01b40ea83c4747612a6